File: ios/ios/GraniteCoreModule.m
//
//  GraniteCoreModule.m
//  ios
//
//  Objective-C bridge for GraniteCoreModule Swift implementation
//  Handles event system and lazy loading functionality
//

#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(GraniteCoreModule, NSObject)

// Event System Methods
RCT_EXTERN_METHOD(addListener:(NSString *)eventName)

RCT_EXTERN_METHOD(removeListeners:(nonnull NSNumber *)count)

// Lazy Loading Method
RCT_EXTERN_METHOD(importLazy:(RCTPromiseResolveBlock)resolve
                 rejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/ios/GraniteCoreModule.swift
//
//  GraniteCoreModule.swift
//  ios
//
//  Granite Core TurboModule implementation for iOS
//  Handles event system and lazy loading functionality
//

import Foundation
import React

@objc(GraniteCoreModule)
class GraniteCoreModule: NSObject, RCTBridgeModule {
    
    @objc public static func moduleName() -> String! {
        return "GraniteCoreModule"
    }
    
    @objc public static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    // MARK: - Event System Methods
    
    @objc public func addListener(_ eventName: String) {
        // Event listener management for Granite event system
        // This is required for RCTEventEmitter compatibility
    }
    
    @objc public func removeListeners(_ count: NSNumber) {
        // Remove event listeners for Granite event system
        // This is required for RCTEventEmitter compatibility
    }
    
    @objc public func importLazy(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        // Lazy loading implementation for Granite modules
        DispatchQueue.main.async {
            // Perform any lazy initialization here
            resolve(nil)
        }
    }
}